src/utils/clock/DefaultClock.js

```
@@ -49,38 +45,53 @@ export default class DefaultClock extends EventEmitter {
    * Register a listener for the clock. When it ticks, the
    * clock will provide the time from the configured endpoint
    *
-   * @param listener
-   * @returns {function} a function for deregistering the provided listener
+   * @override
+   * @param {string | symbol} event the event to listen for
+   * @param {Function} fn the function to call when the event is emitted
+   * @param {*} [context] the context to use for the function call
+   * @returns {this} a function for deregistering the provided listener
    */
-  on(event) {
-    let result = super.on.apply(this, arguments);
+  on(event, fn, context) {
+    super.on(event, fn, context);
 
     if (this.listeners(event).length === 1) {
       this.start();
     }
 
-    return result;
+    return this;
   }
 
   /**
    * Register a listener for the clock. When it ticks, the
    * clock will provide the current local system time
    *
-   * @param listener
-   * @returns {function} a function for deregistering the provided listener
+   * @override
+   * @param {string | symbol} event the event to listen for
+   * @param {Function} [fn] the function to call when the event is emitted
+   * @param {*} [context] the context to use for the function call
+   * @param {boolean} [once]
+   * @returns {this}
    */
-  off(event) {
-    let result = super.off.apply(this, arguments);
+  off(event, fn, context, once) {
+    super.off(event, fn, context, once);
 
     if (this.listeners(event).length === 0) {
       this.stop();
     }
 
-    return result;
+    return this;
+  }
+
+  stop() {
+    // Not implemented
```

should we throw an exception here if it's called?